packages/conventional-changelog-cli/README.md

```diff
@@ -57,10 +56,33 @@ To fully customize the tool, please checkout [conventional-changelog](https://gi
 
 The reason why you should commit and tag after `conventionalChangelog` is that the CHANGELOG should be included in the new release, hence `gitRawCommitsOpts.from` defaults to the latest semver tag.
 
-If you use `npm version`, it auto tags immediately after changing the version in package.json. In such case, you might want to specify the version manually and generate the changelog before `npm version`.
 
-Please use this [gist](https://gist.github.com/stevemao/280ef22ee861323993a0) to make a release or change it to your needs.
+## Recommended workflow using `npm version`
+Using the npm scripts to our advantage.
+With the following hooks:
+```json
+{
+  "scripts": {
+    "version": "conventional-changelog -p angular -i CHANGELOG.md -s -r 0 && git add CHANGELOG.md"
+  }
+}
+```
+You could follow the following workflow 
 
+1. Make changes
+2. Commit those changes
+3. Pull all the tags
+4. Run the [`npm version [path|minor|major]`](https://docs.npmjs.com/cli/version) command
+5. Push
```

Comment:
I think you still have to execute `git commit` to commit the modified `CHANGELOG.md` file.

Reply:
`nvm version` run `git commit` for you, that the whole goal here

Reply:
The `npm version` documentation seems to imply that the author of a `version` script should add any files using `git commit` - `Again, scripts should explicitly add generated files to the commit using git add.`

So `npm version` may be adding files, but it seems like npm expects users to do that themselves.

Reply:
yes need to be explicitly added (not committed), that's what I'm doing in the `version` hook.

That's what I use in a daily bases on several repos.

Reply:
Thank you for the clarification @micky2be!